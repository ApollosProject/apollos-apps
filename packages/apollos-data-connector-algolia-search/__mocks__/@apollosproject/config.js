const ApolloServer = require.requireActual('@apollosproject/config').default;

ApolloServer.loadJs({
  ROCK: {
    TIMEZONE: 'America/New_York',
  },
  ALGOLIA: {
    APPLICATION_ID: 'some-application-id',
    API_KEY: 'some-api-key',
    CONFIGURATION: {},
  },
});

export default ApolloServer;
