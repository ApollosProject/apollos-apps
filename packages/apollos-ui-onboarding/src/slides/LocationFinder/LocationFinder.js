import React, { memo } from 'react';
import PropTypes from 'prop-types';

import {
  CampusCard,
  PaddedView,
  styled,
  Button,
  Touchable,
} from '@apollosproject/ui-kit';

import Slide, { SlideContent } from '../../Slide';

const getCampusAddress = (campus) =>
  `${campus.street1}\n${campus.city}, ${campus.state} ${campus.postalCode}`;

const StyledCampusCard = styled(({ theme }) => ({
  marginBottom: theme.sizing.baseUnit,
}))(CampusCard);

const StyledSlideContent = styled({
  flex: 1,
  justifyContent: 'space-between',
})(SlideContent);

// memo = sfc PureComponent ðŸ’¥
const LocationFinder = memo(
  ({
    onPressPrimary,
    BackgroundComponent,
    slideTitle,
    description,
    buttonText,
    buttonDisabled,
    onPressButton,
    isCurrentCampus,
    campus,
    ...props
  }) => (
    <Slide {...props}>
      {BackgroundComponent}
      <StyledSlideContent title={slideTitle} description={description}>
        {campus ? (
          <Touchable onPress={onPressButton}>
            <StyledCampusCard
              key={campus.id}
              distance={campus.distanceFromLocation}
              title={campus.name}
              description={getCampusAddress(campus)}
              images={[campus.image]}
            />
          </Touchable>
        ) : (
          <PaddedView horizontal={false}>
            <Button
              title={buttonText}
              onPress={onPressButton}
              disabled={buttonDisabled}
              pill={false}
            />
          </PaddedView>
        )}
      </StyledSlideContent>
    </Slide>
  )
);

LocationFinder.CampusCard = StyledCampusCard;
LocationFinder.SlideContent = StyledSlideContent;

LocationFinder.propTypes = {
  /* The `Swiper` component used in `<Onboarding>` looks for and hijacks the title prop of it's
   * children. Thus we have to use a more unique name.
   */
  onPressPrimary: PropTypes.func,
  BackgroundComponent: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  slideTitle: PropTypes.string,
  description: PropTypes.string,
  buttonText: PropTypes.string,
  buttonDisabled: PropTypes.bool,
  onPressButton: PropTypes.func,
  campus: PropTypes.shape({
    image: PropTypes.shape({
      uri: PropTypes.string,
    }),
    distanceFromLocation: PropTypes.number,
    id: PropTypes.string,
    name: PropTypes.string,
  }),
  isCurrentCampus: PropTypes.bool,
};

LocationFinder.displayName = 'LocationFinder';

LocationFinder.defaultProps = {
  slideTitle: "Let's select your local campus",
  description:
    "We'll use your location to connect you with your nearby campus and community",
  buttonText: 'Yes, find my local campus',
  buttonDisabled: false,
};

export default LocationFinder;
