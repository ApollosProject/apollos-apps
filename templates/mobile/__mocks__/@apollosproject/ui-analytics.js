export const withTrackOnPress = (Component) => (props) =>
  <Component onPress={() => null} {...props} />;
export const AnalyticsConsumer = ({ children }) =>
  children({ track: () => null, notify: () => null });
