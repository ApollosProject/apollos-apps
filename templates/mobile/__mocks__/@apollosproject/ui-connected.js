import { gql } from '@apollo/client';
import ApollosConfig from '@apollosproject/config';

export const GET_USER_PROFILE = gql`
  query CurrentUserProfile {
    currentUser {
      id
      profile {
        ...UserProfileParts
        campus {
          ...CampusParts
        }
      }
    }
  }
  ${ApollosConfig.FRAGMENTS.CAMPUS_PARTS_FRAGMENT}
  ${ApollosConfig.FRAGMENTS.USER_PROFILE_PARTS_FRAGMENT}
`;

export const UserAvatarUpdate = 'UserAvatarUpdate';
export const usePersonFollowing = () => ({ total: 1 });
export const usePersonFollowedBy = () => ({ total: 1 });
export const useCurrentUserFollowRequests = () => ({ total: 1 });
