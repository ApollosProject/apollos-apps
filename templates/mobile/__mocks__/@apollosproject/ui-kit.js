export const named = () => (Component) => (props) => <Component {...props} />;
export const styled = (style) => (Component) => (props) =>
  <Component {...props} style={style} />;
export const withTheme = (newProps) => (Component) => (props) =>
  <Component {...props} {...newProps} />;
export const H3 = 'H3';
export const H5 = 'H5';
export const H6 = 'H6';
export const PaddedView = 'PaddedView';
export const BackgroundView = 'BackgroundView';
export const TableView = 'TableView';
export const Row = 'Row';
export const Divider = 'Divider';
export const Touchable = 'Touchable';
export const TextInput = 'TextInput';
export const Button = 'Button';
export const NavigationService = { resetToAuth: () => null };
