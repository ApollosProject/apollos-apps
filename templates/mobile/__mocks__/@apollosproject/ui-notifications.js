import { createContext } from 'react';

export const requestPermissions = () => null;
export const PushContext = createContext({
  hasPrompted: true,
  hasPushPermission: true,
  loading: true,
  checkPermissions: () => {}, // deprecated, confusing name
  updatePermissionStatus: () => {},
});
