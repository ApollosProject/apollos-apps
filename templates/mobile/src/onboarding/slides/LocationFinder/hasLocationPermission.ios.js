// TODO this would be better than getting the position to check
// if we could figure out a way to make it happen
// react-native-permissions can do it, but that requires allowing ALL
// permissions in the app
export default null;
