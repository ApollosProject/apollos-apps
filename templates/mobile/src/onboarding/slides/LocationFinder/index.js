import LocationFinder from './LocationFinder';

export LocationFinderConnected from './LocationFinderConnected';
export default LocationFinder;
