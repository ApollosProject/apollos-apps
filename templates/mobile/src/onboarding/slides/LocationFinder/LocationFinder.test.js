import renderer from 'react-test-renderer';

import { GradientOverlayImage } from '@apollosproject/ui-kit';

import LocationFinder from './LocationFinder';

const campus = {
  id: 'Campus:a0f64573eabf00a607bec911794d50fb',
  name: 'Chicago Campus',
  latitude: 42.09203,
  longitude: -88.13289,
  distanceFromLocation: null,
  street1: '67 Algonquin Rd',
  street2: '',
  city: 'South Barrington',
  state: 'IL',
  postalCode: '60010-6143',
  image: {
    uri: 'https://picsum.photos/300/300/?random',
  },
};

describe('The Onboarding LocationFinder component', () => {
  it('should render', () => {
    const tree = renderer.create(<LocationFinder />);
    expect(tree).toMatchSnapshot();
  });
  it('should render a BackgroundComponent', () => {
    const tree = renderer.create(
      <LocationFinder
        BackgroundComponent={
          <GradientOverlayImage
            source={'https://picsum.photos/640/640/?random'}
          />
        }
      />
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a custom title', () => {
    const tree = renderer.create(
      <LocationFinder slideTitle={'Custom title text'} />
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a custom description', () => {
    const tree = renderer.create(
      <LocationFinder description={'Custom description text'} />
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with a button', () => {
    const tree = renderer.create(<LocationFinder onPressButton={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });
  it('should render the campus card', () => {
    const tree = renderer.create(
      <LocationFinder campus={campus} onPressPrimary={() => jest.fn()} />
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a button with custom text', () => {
    const tree = renderer.create(
      <LocationFinder
        onPressButton={jest.fn()}
        buttonText={'Custom button text'}
      />
    );
    expect(tree).toMatchSnapshot();
  });
  it('should pass additional props to Slide component', () => {
    const tree = renderer.create(
      <LocationFinder onPressSecondary={jest.fn()} secondaryNavText={'Later'} />
    );
    expect(tree).toMatchSnapshot();
  });
});
