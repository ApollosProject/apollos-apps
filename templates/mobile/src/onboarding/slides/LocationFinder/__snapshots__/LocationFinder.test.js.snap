// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Onboarding LocationFinder component should pass additional props to Slide component 1`] = `
Array [
  <RCTScrollView
    contentContainerStyle={
      Object {
        "minHeight": "100%",
      }
    }
    overScrollMode="auto"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <SafeAreaView
        style={[Function]}
      >
        <View>
          <View
            style={[Function]}
          >
            <H2
              style={[Function]}
            >
              Let's select your local campus
            </H2>
          </View>
          <H5
            style={[Function]}
          >
            We'll use your location to connect you with your nearby campus and community
          </H5>
        </View>
        <PaddedView
          horizontal={false}
        >
          <Button
            pill={false}
            title="Yes, find my local campus"
          />
        </PaddedView>
      </SafeAreaView>
    </View>
  </RCTScrollView>,
  <PaddedView
    style={[Function]}
    vertical={false}
  >
    <SafeAreaView
      edges={
        Array [
          "bottom",
        ]
      }
    >
      <ButtonLink
        onPress={[MockFunction]}
        style={[Function]}
      >
        Later
      </ButtonLink>
    </SafeAreaView>
  </PaddedView>,
]
`;

exports[`The Onboarding LocationFinder component should render 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "minHeight": "100%",
    }
  }
  overScrollMode="auto"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <SafeAreaView
      style={[Function]}
    >
      <View>
        <View
          style={[Function]}
        >
          <H2
            style={[Function]}
          >
            Let's select your local campus
          </H2>
        </View>
        <H5
          style={[Function]}
        >
          We'll use your location to connect you with your nearby campus and community
        </H5>
      </View>
      <PaddedView
        horizontal={false}
      >
        <Button
          pill={false}
          title="Yes, find my local campus"
        />
      </PaddedView>
    </SafeAreaView>
  </View>
</RCTScrollView>
`;

exports[`The Onboarding LocationFinder component should render a BackgroundComponent 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "minHeight": "100%",
    }
  }
  overScrollMode="auto"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <GradientOverlayImage
      source="https://picsum.photos/640/640/?random"
    />
    <SafeAreaView
      style={[Function]}
    >
      <View>
        <View
          style={[Function]}
        >
          <H2
            style={[Function]}
          >
            Let's select your local campus
          </H2>
        </View>
        <H5
          style={[Function]}
        >
          We'll use your location to connect you with your nearby campus and community
        </H5>
      </View>
      <PaddedView
        horizontal={false}
      >
        <Button
          pill={false}
          title="Yes, find my local campus"
        />
      </PaddedView>
    </SafeAreaView>
  </View>
</RCTScrollView>
`;

exports[`The Onboarding LocationFinder component should render a button with custom text 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "minHeight": "100%",
    }
  }
  overScrollMode="auto"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <SafeAreaView
      style={[Function]}
    >
      <View>
        <View
          style={[Function]}
        >
          <H2
            style={[Function]}
          >
            Let's select your local campus
          </H2>
        </View>
        <H5
          style={[Function]}
        >
          We'll use your location to connect you with your nearby campus and community
        </H5>
      </View>
      <PaddedView
        horizontal={false}
      >
        <Button
          onPress={[MockFunction]}
          pill={false}
          title="Custom button text"
        />
      </PaddedView>
    </SafeAreaView>
  </View>
</RCTScrollView>
`;

exports[`The Onboarding LocationFinder component should render a custom description 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "minHeight": "100%",
    }
  }
  overScrollMode="auto"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <SafeAreaView
      style={[Function]}
    >
      <View>
        <View
          style={[Function]}
        >
          <H2
            style={[Function]}
          >
            Let's select your local campus
          </H2>
        </View>
        <H5
          style={[Function]}
        >
          Custom description text
        </H5>
      </View>
      <PaddedView
        horizontal={false}
      >
        <Button
          pill={false}
          title="Yes, find my local campus"
        />
      </PaddedView>
    </SafeAreaView>
  </View>
</RCTScrollView>
`;

exports[`The Onboarding LocationFinder component should render a custom title 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "minHeight": "100%",
    }
  }
  overScrollMode="auto"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <SafeAreaView
      style={[Function]}
    >
      <View>
        <View
          style={[Function]}
        >
          <H2
            style={[Function]}
          >
            Custom title text
          </H2>
        </View>
        <H5
          style={[Function]}
        >
          We'll use your location to connect you with your nearby campus and community
        </H5>
      </View>
      <PaddedView
        horizontal={false}
      >
        <Button
          pill={false}
          title="Yes, find my local campus"
        />
      </PaddedView>
    </SafeAreaView>
  </View>
</RCTScrollView>
`;

exports[`The Onboarding LocationFinder component should render the campus card 1`] = `
Array [
  <RCTScrollView
    contentContainerStyle={
      Object {
        "minHeight": "100%",
      }
    }
    overScrollMode="auto"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <SafeAreaView
        style={[Function]}
      >
        <View>
          <View
            style={[Function]}
          >
            <H2
              style={[Function]}
            >
              Let's select your local campus
            </H2>
          </View>
          <H5
            style={[Function]}
          >
            We'll use your location to connect you with your nearby campus and community
          </H5>
        </View>
        <Touchable>
          <CampusCard
            description="67 Algonquin Rd
South Barrington, IL 60010-6143"
            images={
              Array [
                Object {
                  "uri": "https://picsum.photos/300/300/?random",
                },
              ]
            }
            style={[Function]}
            title="Chicago Campus"
          />
        </Touchable>
      </SafeAreaView>
    </View>
  </RCTScrollView>,
  <PaddedView
    style={[Function]}
    vertical={false}
  >
    <SafeAreaView
      edges={
        Array [
          "bottom",
        ]
      }
    >
      <Button
        onPress={[Function]}
        style={Object {}}
      >
        <H5>
          Next
        </H5>
        <Icon
          name="arrow-next"
        />
      </Button>
    </SafeAreaView>
  </PaddedView>,
]
`;

exports[`The Onboarding LocationFinder component should render with a button 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "minHeight": "100%",
    }
  }
  overScrollMode="auto"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <SafeAreaView
      style={[Function]}
    >
      <View>
        <View
          style={[Function]}
        >
          <H2
            style={[Function]}
          >
            Let's select your local campus
          </H2>
        </View>
        <H5
          style={[Function]}
        >
          We'll use your location to connect you with your nearby campus and community
        </H5>
      </View>
      <PaddedView
        horizontal={false}
      >
        <Button
          onPress={[MockFunction]}
          pill={false}
          title="Yes, find my local campus"
        />
      </PaddedView>
    </SafeAreaView>
  </View>
</RCTScrollView>
`;
